docs: update the API docs (#6565)
apache/apisix: docs/en/latest/admin-api.md

**Note**: Mentions of `X-API-KEY` in this document refers to `apisix.admin_key.key`—the access token for Admin API—in your configuration file.

## Route

**API**: /apisix/admin/routes/{id}?ttl=0

[Routes](./architecture-design/route.md) match the client's request based on defined rules, loads and executes the corresponding [plugins](#plugin), and forwards the request to the specified [Upstream](#upstream).

**Note**: When the Admin API is enabled, to avoid conflicts with your design API, use a different port for the Admin API. This can be set in your configuration file by changing the `port_admin` key.

### Request Methods

| Method | Request URI                      | Request Body | Description                                                                                                                   |
| ------ | -------------------------------- | ------------ | ----------------------------------------------------------------------------------------------------------------------------- |

> | GET    | /apisix/admin/routes             | NULL         | Fetches a list of all configured Routes. list                                                                                 |

Reviewer: The ending `list` is redundant?

Author: @moonming Using a list keeps it consistent with other items in the page even though there is only one item. It was a single item list before as well and I did not change it.

Should I move it out of the list or keep it as it is?

Reviewer 2: I would prefer removing the `list` currently because your description is clearer. 😉

Author: Oh. I misread the message and did not see the actual `list` word. I will remove that.

| GET    | /apisix/admin/routes/{id}        | NULL         | Fetches specified Route by id.                                                                                                |
| PUT    | /apisix/admin/routes/{id}        | {...}        | Creates a Route with the specified id.                                                                                            |

> | POST   | /apisix/admin/routes             | {...}        | Creates a Route and assigns an id.                                                                                            |

Reviewer 2: ```suggestion
| POST   | /apisix/admin/routes             | {...}        | Creates a Route and assigns an random id.                                                                                            |
```

Author: Sure, I will make this change throughout.

### Response Parameters

Currently, the response is returned from etcd.

[Back to TOC](#table-of-contents)

## Service

**API**: /apisix/admin/services/{id}

A Service is an abstraction of an API (which can also be understood as a set of Route abstractions). It usually corresponds to an upstream service abstraction.

The relationship between Routes and a Service is usually N:1.

### Request Methods

| Method | Request URI                        | Request Body | Description                                                                                                                     |
| ------ | ---------------------------------- | ------------ | ------------------------------------------------------------------------------------------------------------------------------- |
| GET    | /apisix/admin/services             | NULL         | Fetches a list of available Services.                                                                                           |
| GET    | /apisix/admin/services/{id}        | NULL         | Fetches specified Service by id.                                                                                                |

> | PUT    | /apisix/admin/services/{id}        | {...}        | Creates a Service with specified id.                                                                                            |

Reviewer 3: ```suggestion
| PUT    | /apisix/admin/services/{id}        | {...}        | Creates a Service with a specified id.                                                                                            |
```

> | POST   | /apisix/admin/services             | {...}        | Creates a Service and assigns an id.                                                                                            |

Reviewer 2: How about using `random id` and `specified id`?

Author: +1

| ----------- | -------- | ----------- | ------------------------------------------------------------------------------------------------------------------ | ------------------------------------------------ |
| username    | True     | Name        | Name of the Consumer.                                                                                              |                                                  |
| plugins     | False    | Plugin      | Plugins that are executed during the request/response cycle. See [Plugin](architecture-design/plugin.md) for more. |                                                  |
| desc        | False    | Auxiliary   | Description of usage scenarios.                                                                                    | customer xxxx                                    |
| labels      | False    | Match Rules | Attributes of the Consumer specified as key-value pairs.                                                           | {"version":"v2","build":"16","env":"production"} |

> | create_time | False    | Auxiliary   | Epoch timestamp in second of the created time. If missing, this field will be populated automatically.             | 1602883670                                       |

Reviewer 4: > Epoch timestamp in second of the created time.

I am bit confused on the framing of this sentence. How about this -

```suggestion
| create_time | False    | Auxiliary   | Epoch timestamp (in seconds) of the created time. If missing, this field will be populated automatically.             | 1602883670                                       |
```

Author: Yes. I missed this.

}
```

> When bound to Routes or Services, the authentication Plugins infers the Consumer from the request and does not require any parameters. Whereas, while it is bound to a Consumer, username, password and other information needs to be provided.

Reviewer 4: ```suggestion
When bound to a Route or Service, the Authentication plugin infers the Consumer from the request and does not require any parameters. Whereas, when bound to a Consumer, username, password and other information needs to be provided.
```
